src/expr/src/vector_op/cast.rs

```diff
@@ -178,6 +186,77 @@ pub fn i64_to_timestampz(t: i64) -> Result<i64> {
     }
 }
 
+#[inline(always)]
+pub fn str_to_bytea(elem: &str) -> Result<Bytes> {
+    // Padded with whitespace str is not allowed.
+    if elem.starts_with(' ') && elem.trim().starts_with("\\x") {
+        Err(ExprError::Parse(PARSE_ERROR_STR_TO_BYTEA))
+    } else if let Some(remainder) = elem.strip_prefix(r"\x") {
+        Ok(parse_bytes_hex(remainder)?.into())
+
+        // Ok(hex::decode(remainder).unwrap().into())
+    } else {
+        // use hex::ToHex;
+        // Ok(elem.encode_hex::<String>().into())
+        // Ok(hex::decode(elem).unwrap().into())
+        Ok(parse_bytes_traditional(elem).unwrap().into())
+    }
+}
+
+pub fn parse_bytes_hex(s: &str) -> Result<Vec<u8>> {
+    // Can't use `hex::decode` here, as it doesn't tolerate whitespace
+    // between encoded bytes.
+
+    let decode_nibble = |b| match b {
+        b'a'..=b'f' => Ok(b - b'a' + 10),
+        b'A'..=b'F' => Ok(b - b'A' + 10),
+        b'0'..=b'9' => Ok(b - b'0'),
+        _ => Err(ExprError::Parse(PARSE_ERROR_STR_TO_BYTEA)),
+    };
+
+    let mut buf = vec![];
+    let mut nibbles = s.as_bytes().iter().copied();
+    while let Some(n) = nibbles.next() {
+        if let b' ' | b'\n' | b'\t' | b'\r' = n {
+            continue;
+        }
+        let n = decode_nibble(n)?;
+        let n2 = match nibbles.next() {
+            // None => return Err(ParseHexError::OddLength),
+            None => return Err(ExprError::Parse(PARSE_ERROR_STR_TO_BYTEA)),
+            // None => todo!(),
+            Some(n2) => decode_nibble(n2)?,
+        };
+        buf.push((n << 4) | n2);
+    }
+    Ok(buf)
+}
+
+pub fn parse_bytes_traditional(s: &str) -> Result<Vec<u8>> {
```

Comment:
Function `parse_bytes_traditional` and `parse_bytes_hex` are copied from Mz and it works very well. However I wonder whether the license will be a problem?

Reply:
cc @neverchanje